js/packages/phoenix-client/src/schemas/llm/messageSchemas.ts

```diff
@@ -442,6 +493,124 @@ export const openAIMessageToPrompt = openAIMessageSchema.transform(
+                ""),
+        };
+      case "user":
+        // take text and image parts, ignore other parts
+        return {
+          role: "user",
+          content:
+            typeof openai.content === "string"
+              ? openai.content
+              : openai.content
+                  .filter(
+                    (part) => part.type === "text" || part.type === "image_url"
+                  )
+                  .map((part) => {
+                    if (part.type === "text") {
+                      return {
+                        type: "text",
+                        text: part.text,
+                      };
+                    }
+                    if (part.type === "image_url") {
+                      return {
+                        type: "image",
+                        image: part.image_url.url,
+                      };
+                    }
+
+                    return assertUnreachable(part);
+                  }),
+        };
+      case "assistant": {
+        type AssistantMessage = Extract<VercelAIMessage, { role: "assistant" }>;
+        // take text any parts, convert tool calls to tool call parts, ignore other parts
+        const newContent: AssistantMessage["content"] = [];
+        // take all text parts from openai message
+        if (typeof openai.content === "string") {
+          newContent.push({ type: "text", text: openai.content });
+        } else {
+          openai.content.forEach((part) => {
+            if (part.type === "text") {
+              newContent.push({ type: "text", text: part.text });
+            }
+          });
+        }
+        // add any tool calls
+        if (openai.tool_calls) {
+          openai.tool_calls.forEach((tc) => {
+            newContent.push({
+              type: "tool-call",
+              toolCallId: tc.id,
+              toolName: tc.function.name,
+              args: tc.function.arguments,
+            });
+          });
+        }
+        return {
+          role: "assistant",
+          content: newContent,
+        };
+      }
+      case "tool": {
+        type ToolMessage = Extract<VercelAIMessage, { role: "tool" }>;
+        const newContent: ToolMessage["content"] = [];
+        if (typeof openai.content === "string") {
+          newContent.push({
+            type: "tool-result",
+            toolCallId: openai.tool_call_id,
+            toolName: "", // We don't have this??
+            result: openai.content,
+          });
+        } else {
+          openai.content.forEach((part) => {
+            if (part.type === "text") {
+              newContent.push({
+                type: "tool-result",
+                toolCallId: openai.tool_call_id,
+                toolName: "", // We don't have this??
+                result: part.text,
+              });
+              return;
+            }
+            assertUnreachable(part.type);
+          });
+        }
```

Comment:
I'm not sure what to do about these missing toolName fields.

I assume vercel fills them in when you execute tools through the sdk, and things seem to work without it, but the types say this field is required.

Reply:
My guess is that it's actually optional, and just used if you provide an `execute` function with your tool definition.

https://sdk.vercel.ai/docs/ai-sdk-core/tools-and-tool-calling